static/js/snap-metrics.js

```diff
@@ -0,0 +1,251 @@
+   */
+  function debounce(func, wait, immediate) {
+    var timeout;
+    return function() {
+      var context = this, args = arguments;
+      var later = function() {
+        timeout = null;
+        if (!immediate) func.apply(context, args);
+      };
+      var callNow = immediate && !timeout;
+      clearTimeout(timeout);
+      timeout = setTimeout(later, wait);
+      if (callNow) func.apply(context, args);
+    };
+  }
+
+  /**
+   * Render all metrics
+   * @param {Object} metrics An object of metrics from the API.
+   */
+  function renderMetrics(metrics) {
+    if (!d3 || ! bb) {
+      return false;
+    }
+
+    var X_TICK_FREQUENCY = 7;
+    var Y_TICK_FREQUENCY = 5;
+
+    var TICK_START = 4;
+    var DEFAULT_TICK_LENGTH = 8;
+    var KEY_TICK_LENGTH = 16;
+    var TEXT_OFFSET = 28;
+
+    var COLORS = {
+      installs: '#94519E'
+    };
+
+    /**
+     * Cull Y Axis. The built in culling does not provide enough control.
+     *  - hide labels that are not every X_TICK_FREQUENCY ticks.
+     *  - remove month abreviation from label for sequential dates that have 
+     *    the same month.
+     * @param {NodeList} ticks X axis tick elements.
+     */
+    function cullXAxis(ticks) {
+      var tick, totalTicks, text, monthCache;
+
+      for (tick = 0, totalTicks = ticks.length; tick < totalTicks; tick += 1) {
+        text = ticks[tick].querySelector('text');
+
+        if (tick % X_TICK_FREQUENCY !== 0) {
+          text.style.display = 'none';
+        } else {
+          ticks[tick].classList.add('active');
+          text.children[0].setAttribute('fill', '#000');
+          var month = text.children[0].innerHTML.split(' ');
+          if (month[0] === monthCache) {
+            text.children[0].innerHTML = month[1];
+          }
+          monthCache = month[0];
+        }
+      }
+    }
+
+    /**
+     * Cull Y Axis. The built in culling does not provide enough control.
+     *  - hide labels that are not every Y_TICK_FREQUENCY ticks.
+     * @param {NodeList} ticks Y axis tick elements.
+     */
+    function cullYAxis(ticks) {
+      var tick, totalTicks, text;
+
+      for (tick = 0, totalTicks = ticks.length; tick < totalTicks; tick += 1) {
+        text = ticks[tick].querySelector('text');
+
+        if (tick % Y_TICK_FREQUENCY !== 0) {
+          text.style.display = 'none';
+        } else {
+          ticks[tick].classList.add('active');
+        }
+      }
+    }
+
+    /**
+     * Update graph x and y axis formatting.
+     * @param {HTMLElement} el Graph wrapping element.
+     */
+    function formatAxis(el) {
+      var xAxis = el.querySelector('.bb-axis-x');
+      
+      var ticks = xAxis.querySelectorAll('.tick');
+      cullXAxis(ticks);
+
+      var yAxis = el.querySelector('.bb-axis-y');
+
+      ticks = yAxis.querySelectorAll('.tick');
+      cullYAxis(ticks);
+    }
+
+    /**
+     * Generate the tooltip.
+     * @param {Object} data The point data.
+     * @returns {HTMLElement}
```

Comment:
It doesn't seem to return an element but a HTML string.

Reply:
Good spot, I'm a LIAR!

Reply:
dun